sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -569,21 +576,25 @@ class CompactBlockProcessor internal constructor(
                 is SyncingResult.RestartSynchronization -> {
                     // Restarting the synchronization process
                     return BlockProcessingResult.RestartSynchronization
-                } else -> {
-                    // An error came - remove persisted but not scanned blocks
-                    val lastScannedHeight = when (val result = getMaxScannedHeight(backend)) {
-                        is GetMaxScannedHeightResult.Success -> result.height
-                        else -> null
-                    }
-                    lastScannedHeight?.let {
-                        downloader.rewindToHeight(lastScannedHeight)
-                    }
+                }
+                is SyncingResult.ContinuityError -> {
+                    val failedHeight = (syncingResult as SyncingResult.ContinuityError).failedAtHeight
+                    checkContinuityErrorResult(failedHeight)
+                    // This step is independent of the rewind action as it only removes temporary persisted block files
+                    // from the device storage. The files will be re-downloaded in the following synchronization cycle.
```

Comment:
Can you point to where the rewind will be called?

Reply:
There is `handleChainError` in the `checkContinuityErrorResult`. https://github.com/zcash/zcash-android-wallet-sdk/blob/613c951282b692bc6c9836198808d1668a591e59/sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt#L1820

Reply:
Aha, okay, I didn't expand `checkContinuityErrorResult`. Thanks. The naming there is a little confusing, because it's actually handling the continuity error, not just checking for it.

Reply:
Yeah, understand. What about renaming `checkContinuityErrorResult` to `processContinuityError`?